owncloudApp/src/main/java/com/owncloud/android/usecases/transfers/uploads/UploadFileFromContentUriUseCase.kt

```diff
@@ -52,13 +58,20 @@ class UploadFileFromContentUriUseCase(
             .build()
 
         val uploadFileFromContentUriWorker = OneTimeWorkRequestBuilder<UploadFileFromContentUriWorker>()
-            .setInputData(inputData)
+            .setInputData(inputDataUploadFileFromContentUriWorker)
             .setConstraints(constraints)
             .addTag(params.accountName)
             .addTag(params.uploadIdInStorageManager.toString())
             .build()
 
-        workManager.enqueue(uploadFileFromContentUriWorker)
+        val removeLocalFileWorker = OneTimeWorkRequestBuilder<RemoveLocalFileWorker>()
+            .setInputData(inputDataRemoveLocalFileWorker)
+            .build()
+
+        workManager.beginWith(uploadFileFromContentUriWorker)
+            .then(removeLocalFileWorker) // File is already uploaded, so the original one can be removed if the behaviour is MOVE
+            .enqueue()
```

Comment:
Instead of starting a new worker, we could save it when not needed by checking here the behaviour instead of sending it to the new worker

Reply:
Every invocation of the `WorkContinuation.then(...)` returns a new instance of `WorkContinuation`. So we need to make sure we are calling `enqueue() `on the correct instance. 
`workContinuation` represents a continuation of work and should be modified to include the new task (in this case, removeLocalFileWorker) only if a specific condition is met (behavior == UploadBehavior.MOVE). Therefore, we update `workContinuation` with the new instance:
`workContinuation = workContinuation.then(removeLocalFileWorker)`